openapi/generator.go

```diff
@@ -1161,3 +1169,19 @@ func fieldNameFromTag(sf reflect.StructField, tagName string) string {
 	}
 	return name
 }
+
+/// parseExampleValue is used to transform the string representation of the example value to the correct type.
+func parseExampleValue(schemaType, value string) (interface{}, error) {
+	switch schemaType {
+	case "boolean":
+		return strconv.ParseBool(value)
+	case "string":
+		return value, nil
+	case "integer":
```

Comment:
`strconv.Atoi` is too naive here. What about `uint8` fields that would provide an oveflowing example value ? We need to use `strconv.ParseInt` or `strconv.ParseUint` with the proper `bitSize`. If an error occures, we return it and leave the caller decide what to do (ignore the example value and collect the error).